File: src/main/java/com/eh2/noframe/domain/BookEntity.java
package com.eh2.noframe.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class BookEntity implements Cloneable {

	private String id;

	private String title;

	public BookEntity(String title) {
		this.title = title;
	}

	private void setId(String id) {
		this.id = id;
	}

	public BookEntity generateWithId(String id) throws CloneNotSupportedException {
		BookEntity clone = (BookEntity) this.clone();
		((BookEntity) clone).setId(id);
		return clone;
	}

}

File: src/main/kotlin/com/eh2/noframe/domain/BookEntity.kt
package com.eh2.noframe.domain

class BookEntity(
    val id: String?,
    val title: String
) : Cloneable {

    constructor(title: String) : this(null, title)

    @Throws(CloneNotSupportedException::class)
    fun generateWithId(id: String): BookEntity {
        return BookEntity(id, title)
    }
}
